packages/fx-core/src/core/FxCoreImplementV3.ts

```diff
@@ -663,15 +669,27 @@ export class FxCoreV3Implement {
   @hooks([ErrorHandlerMW, QuestionMW("copilotPluginAddAPI"), ConcurrentLockerMW])
   async copilotPluginAddAPI(inputs: Inputs): Promise<Result<undefined, FxError>> {
     const operations = inputs[QuestionNames.ApiOperation] as string[];
-    const openapiSpecPath =
-      inputs[QuestionNames.ApiSpecLocation] ?? inputs.openAIPluginManifest?.api.url;
+    const url = inputs[QuestionNames.ApiSpecLocation] ?? inputs.openAIPluginManifest?.api.url;
+    const apiSpecFolderPath = path.join(
+      inputs.projectPath!,
+      AppPackageFolderName,
+      apiSpecFolderName
+    );
+    await fs.ensureDir(apiSpecFolderPath);
+
+    const isYaml = await isYamlSpecFile(url);
+    const openapiSpecPath = path.join(
+      apiSpecFolderPath,
+      isYaml ? apiSpecYamlFileName : apiSpecJsonFileName
+    );
     const manifestPath = inputs[QuestionNames.ManifestPath];
     const specParser = new SpecParser(openapiSpecPath);
     const adaptiveCardFolder = path.join(
       inputs.projectPath!,
       AppPackageFolderName,
       AdaptiveFolderName
     );
+    await fs.ensureDir(adaptiveCardFolder);
```

Comment:
wondering if we need this. I think adaptiveCardFolder will be created in the lib if it does not exist.